src/test/librados/test.cc

```diff
@@ -241,14 +241,29 @@ int destroy_one_pool(const std::string &pool_name, rados_t *cluster)
 int destroy_one_ec_pool(const std::string &pool_name, rados_t *cluster)
 {
   int ret = rados_pool_delete(*cluster, pool_name.c_str());
-  if (ret == 0) {
-    int ret2 = destroy_ec_profile(cluster);
-    if (ret2) {
-      rados_shutdown(*cluster);
-      return ret2;
-    }
-    rados_wait_for_latest_osdmap(*cluster);
+  if (ret) {
+    rados_shutdown(*cluster);
+    return ret;
+  }
+
+  char *cmd[2];
+
+  cmd[0] = (char *)(std::string("{\"prefix\": \"osd crush rule rm\", \"name\":\"") + pool_name + std::string("\"}")).c_str();
```

Comment:
Could this be added in destroy_ec_profile instead ? The only other place this function is called won't be negatively impacted by the addtional crush rule rm. Or is there another reason for inlining it ?